src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiPlugin.groovy

```diff
@@ -422,10 +422,13 @@ class JpiPlugin implements Plugin<Project> {
 
         // generate test hpl manifest for the current plugin, to be used during unit test
         def outputDir = project.layout.buildDirectory.dir('generated-resources/test')
-        project.tasks.register('generate-test-hpl', GenerateTestHpl) {
+        testSourceSet.output.dir(outputDir)
+
+        def generateTestHplTask = project.tasks.register('generate-test-hpl', GenerateTestHpl) {
             it.hplDir.set(outputDir)
         }
-        testSourceSet.output.dir(outputDir)
```

Comment:
Doesn't this configure the output directory and add the dependency automatically:

```
testSourceSet.output.dir(generateHplTask.map { it.hplDir })
```

Reply:
@darxriggs could you also add a test case for this task dependency in JpiIntegrationSpec’s `#task should run #dependency`?

Thanks for hunting this down and sending in the change, and thanks to @wolfs for the review and suggestion.

Reply:
I tried to implement it similar to [configureInjectedTest](https://github.com/jenkinsci/gradle-jpi-plugin/blob/v0.36.0/src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiPlugin.groovy#L297). More feedback is welcome. For me it is also not clear what the best practice is regarding `testSourceSet.java.srcDirs += anotherDir` vs. `testSourceSet.java.srcDirs(anotherDir)`.

Reply:
I will stick with the explicit dependency declaration. The suggested implementation doesn't work with Gradle 4 (we are testing with 4, 5 and 6).

Reply:
I added the test as suggested.